### ISSUE TITLE
Request.Files is empty on Mono Self Host

### ISSUE THREAD

**Issue by scottmcarthur** [ServiceStack/Issues#271]
When using self hosting on Mono, `Request.Files` is always empty.

I am using 4.0.39 with the code I had previously [used here](http://stackoverflow.com/a/21091625/1295344). I have tried Chrome, Safari and Postman, same result.

    using System;
	using ServiceStack;

	namespace Testv4
	{
		class MainClass
		{
			public static void Main()
			{
				var appHost = new AppHost(500);
				appHost.Init();
				appHost.Start("http://*:8082/");
				Console.ReadKey();
			}
		}

		public class TestApp
		{
			[Route("/upload", "POST")]
			public class UploadFileRequest {}

			public class TestController : Service
			{
				public void Any(UploadFileRequest request)
				{
					Console.WriteLine(Request.Files.Length); // Always 0
				}
			}
		}

		public class AppHost : AppHostHttpListenerPoolBase
		{
			public AppHost(int poolSize) : base("Test Service", poolSize, typeof(TestApp).Assembly)
			{
			}

			public override void Configure(Funq.Container container)
			{
			}
		}
	}

The html:

	<!doctype html>
	<html>
	<head>
	    <title>Test</title>
	</head>
	<body>
	<form action="/upload" method="POST" enc="multipart/form-data">
	    <h1>Upload File</h1>
	    <input type="file" name="File1" />
	    <input type="submit" value="Upload" />
	</form>
	</body>
	</html>

**Comment by mythz** [comment]
Hi Scott,

The first issue is that the ContentType attribute should be `enctype` otherwise it comes through as `application/x-www-form-urlencoded`, e.g:

```html
<form action="/upload" method="POST" enctype="multipart/form-data">
    <h1>Upload File</h1>
    <input type="file" name="File1" />
    <input type="submit" value="Upload" />
</form>
```

But there's still a parsing issue as the File count is still 0 which I'm currently debugging - it's just been extremely painful to debug since Xamarin's debugger seems broken.

**Comment by scottmcarthur** [comment]
Thanks for getting back to me on this.

^ Yeah sorry the `enc` instead of `enctype` was just a typo here.

Let me know if there is anything I can do to make it easier to get this resolved. Thanks Demis.

**Comment by mythz** [comment]
Hi Scott, this should now be resolved in the latest v4.0.39 that just finished [deploying to MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

thx for reporting!
